### PR TITLE
FE - Step2: Confirm Seed Time

### PR DESCRIPTION
**Pull request by viriponce**
This PR address issue #132 which is part of #130.

**Note: This PR depends on PR #142 and #138** 

### Implementation

1. New Files/Features

- **frontend/src/GenerateHeats/GenerateHeats.jsx**
  - The existing component `SelectAthletes` was renamed to `GenerateHeats` since this file will contains the components related to each step to Generate Heats: Select Athletes, Confirm Seed Times and Generate Heats.
  - The navigation buttons are conditionally rendered/labeled based on the current step.
  - Added the components to display when user is at Step 2 Confirm Seed Times: 
     - A Generic table listing the athletes selected at Step 1 with their seed time. Each row shows an icon to edit the seed time.
     - When the user clicks the edit button, a popup shows the form to update the athlete's seed time.

- **frontend/src/GenerateHeats/UpdateSeedTime.jsx**
  - This component is introduced in this PR.
  - It handles the rendering of the `SeedTimeForm` and the logic associated to it.
  - It calls the API to save the updated data upon form submission.
  - Triggers the update of the table with the new seed time.
  - It shows a message to let the user knows wether the seed time was updated successfully or not. 
   
3. Updates to Existing Components:

- frontend/src/Event/MeetEventDisplay.jsx
  - Renamed the component SelectAthlete to GenereateHeats.

- frontend/src/components/FormElements/TimeField.jsx
  - Fix the padding for milliseconds. 
  
UI Preview

- After clicking the `Confirm Seed Time` button in the previous step, a list of athletes with their seed time is displayed with the option to edit it:
Screenshot 2024-11-05 at 10 57 56 AM

- Clicking the Edit icon opens a popup form where the user can update the athlete's seed time. Upon submitting changes by clicking the Update button, a success or error message is displayed as appropriate:

success form

 
error form

### CODE REVIEW

File: frontend/src/GenerateHeats/UpdateSeedTime.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  let message = error ? error : "Seed time updated successfully!";
+
+  const handleClearError = () => {
+    console.log("Clear error");
+    setError(false);
+    setSubmitted(false);
+  };
+
+  const onSubmit = async (data) => {
+    setSubmitted(true);
```
Comment (MarisiaS) [inline frontend/src/GenerateHeats/UpdateSeedTime.jsx:38]: Setting submitted triggers the rendering of the alert box. However, when an error occurs, the success alert box appears first, followed by the error alert box.

File: frontend/src/GenerateHeats/GenerateHeats.jsx [hunk 2]
```diff
@@ -78,27 +102,65 @@ const SelectAthlete = ({ eventName, eventId, onBack }) => {
...
+      icon: areAthletesSelected ? <ChecklistIcon /> : <TimeIcon />,
+      onClick: handleCurrentStep,
+      disabled: !areAthletesSelected ? selectedAthletes.length === 0 : false,
     },
     {
       label: "Back to events",
       icon: <BackIcon />,
       onClick: onBack,
     },
+    {
```
Comment (MarisiaS) [inline frontend/src/GenerateHeats/GenerateHeats.jsx:154]: Could you please move this button to the beginning of the list?

File: frontend/src/GenerateHeats/GenerateHeats.jsx [hunk 3]
```diff
@@ -78,27 +102,65 @@ const SelectAthlete = ({ eventName, eventId, onBack }) => {
...
+      label: "Heats",
```
Comment (MarisiaS) [inline frontend/src/GenerateHeats/GenerateHeats.jsx:155]: Change the label to 'Create Heats.' As discussed, the remaining label changes for similar buttons will be addressed in the Generate Heats issue.

**Review by MarisiaS — CHANGES_REQUESTED** [review]

File: frontend/src/GenerateHeats/UpdateSeedTime.jsx (hunk 1 above)
Comment (viriponce) [reply frontend/src/GenerateHeats/UpdateSeedTime.jsx]: Moved setSubmitted after API calling

File: frontend/src/GenerateHeats/GenerateHeats.jsx (hunk 3 above)
Comment (viriponce) [reply frontend/src/GenerateHeats/GenerateHeats.jsx]: Changed the label for the button

File: frontend/src/GenerateHeats/GenerateHeats.jsx (hunk 2 above)
Comment (viriponce) [reply frontend/src/GenerateHeats/GenerateHeats.jsx]: Yes, the Create Heats is now the first one

**Review by MarisiaS — APPROVED** [review]
LGTM

I was able to update the seed time of athletes.
Got the error message when the server was down.
Thanks Viri! Looks great.

**Pull request merged**